pkg/syncer/syncer.go

```diff
@@ -72,6 +74,37 @@ func StartSyncer(ctx context.Context, upstream, downstream *rest.Config, resourc
 	go specSyncer.Start(ctx, numSyncerThreads)
 	go statusSyncer.Start(ctx, numSyncerThreads)
 
+	// TODO(marun) Report pcluster connectivity to kcp
+	kcpClusterClient, err := kcpclient.NewClusterForConfig(upstream)
+	if err != nil {
+		return err
+	}
+	kcpClient := kcpClusterClient.Cluster(kcpClusterName)
+	workloadClustersClient := kcpClient.WorkloadV1alpha1().WorkloadClusters()
+
+	// Attempt to heartbeat every 20s
+	go wait.UntilWithContext(ctx, func(ctx context.Context) {
+		var heartbeatTime time.Time
+
+		// TODO(marun) Figure out a strategy for backoff to avoid a thundering herd problem with lots of syncers
+
+		// Attempt to heartbeat every second until successful. Errors are logged instead of being returned so the
+		// poll error can be safely ignored.
+		_ = wait.PollImmediateInfiniteWithContext(ctx, 1*time.Second, func(ctx context.Context) (bool, error) {
+			patchBytes := []byte(fmt.Sprintf(`[{"op":"replace","path":"/status/lastSyncerHeartbeatTime","value":%q}]`, time.Now().Format(time.RFC3339)))
+			workloadCluster, err := workloadClustersClient.Patch(ctx, pcluster, types.JSONPatchType, patchBytes, metav1.PatchOptions{}, "status")
+			if err != nil {
+				klog.Errorf("failed to set status.lastSyncerHeartbeatTime for WorkloadCluster %s|%s: %v", kcpClusterName, pcluster, err)
+			} else {
+				heartbeatTime = workloadCluster.Status.LastSyncerHeartbeatTime.Time
+			}
+			return err == nil, nil
+		})
+
+		klog.V(5).Infof("Heartbeat set for WorkloadCluster %s|%s: %s", kcpClusterName, pcluster, heartbeatTime)
+
+	}, 20*time.Second) // TODO(marun) Coordinate this value with the interval configured for the heartbeat controller
```

Comment:
Can this at least be a package-level `const` for now? I'm fine with plumbing it through being a later change, it's just hard to discover this TODO as it is.

Reply:
For the heartbeat field itself or for reading the interval in which to heartbeat?

Reply:
I meant for the interval, so it's clear it's a `const` for now that we want to eventually promote into a flag or other config.

Reply:
We definitely don't want it as a flag or any other config - it has to stay consistent with what the heartbeat controller expects.

Reply:
Done.